Mage.Sets/src/mage/cards/p/PushTheLimit.java

```diff
@@ -0,0 +1,138 @@
+        // Vehicles you control become artifact creatures until end of turn. Creatures you control gain haste until end of turn.
+        this.getSpellAbility().addEffect(new VehiclesBecomeArtifactCreatureEffect(Duration.EndOfTurn)
+                .concatBy("<br>"));
+        this.getSpellAbility().addEffect(new GainAbilityAllEffect(HasteAbility.getInstance(), Duration.EndOfTurn, filterCreatures));
+    }
+
+    private PushTheLimit(final PushTheLimit card) {
+        super(card);
+    }
+
+    @Override
+    public PushTheLimit copy() {
+        return new PushTheLimit(this);
+    }
+}
+
+class PushTheLimitSacrificeEffect extends OneShotEffect {
+
+    private final Set<Card> cards;
+
+    public PushTheLimitSacrificeEffect(Set<Card> cards) {
+        super(Outcome.Sacrifice);
+        this.cards = cards;
+    }
+
+    public PushTheLimitSacrificeEffect(final PushTheLimitSacrificeEffect effect) {
+        super(effect);
+        this.cards = effect.cards;
+    }
+
+    @Override
+    public PushTheLimitSacrificeEffect copy() {
+        return new PushTheLimitSacrificeEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Set<Permanent> permanents = cards.stream()
+                .map(Card::getId)
+                .map(game::getPermanent)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toSet());
+        if (permanents.isEmpty()) {
+            return false;
+        }
+        permanents.forEach(permanent -> {
+            permanent.sacrifice(source, game);
+        });
+        return true;
+    }
+}
+
+class PushTheLimitEffect extends ReturnFromYourGraveyardToBattlefieldAllEffect {
+    private static final FilterCard filter = new FilterCard("Mount and Vehicle cards");
+    static {
+        filter.add(Predicates.or(
+                SubType.MOUNT.getPredicate(),
+                SubType.VEHICLE.getPredicate()
+        ));
+    }
+    public PushTheLimitEffect() {
+        super(filter);
+        staticText += ". Sacrifice them at the beginning of the next end step.";
+    }
+
+    public PushTheLimitEffect(final PushTheLimitEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PushTheLimitEffect copy() {
+        return new PushTheLimitEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        Set<Card> cards = controller.getGraveyard().getCards(filter, source.getControllerId(), source, game);
+        boolean result = controller.moveCards(controller.getGraveyard().getCards(filter, source.getControllerId(), source, game),
+                Zone.BATTLEFIELD, source, game, false, false, false, null);
+        if (result) {
+
+            Effect sacrificeEffect = new PushTheLimitSacrificeEffect(cards);
```

Comment:
No needs in custom sacrifice trigger, add directly to permanents. See Push Pull as example:

![IMG_1212](https://github.com/user-attachments/assets/0f85cf0f-3a91-402f-87ff-aaaca85832ed)

Reply:
That incorrectly produces a separate trigger for each creature. I think it'd be best to use a `SacrificeAllControllerEffect` filtered with a `MageObjectReferencePredicate`.

Reply:
Good point. It must generate 1 trigger due rules. So Push // Pull is wrongly implemented.

Reply:
Alternative implementation by SacraficeTargetEffect and fixed targets list:

![IMG_1216](https://github.com/user-attachments/assets/0ae3b214-6bdf-4c14-be26-97073ab685f5)